src/nvim/move.c

```diff
@@ -1349,8 +1361,8 @@ void scroll_cursor_top(int min_scroll, int always)
   linenr_T old_topline = curwin->w_topline;
   linenr_T old_topfill = curwin->w_topfill;
   linenr_T new_topline;
-  assert(p_so <= INT_MAX);
-  int off = (int)p_so;
+  int off = (int)get_scrolloff_value();
+  assert(off <= INT_MAX);
```

Comment:
This assert now doesn't do anything. But probably it should just be deleted, range checks are done when options are _set_, and need not be duplicated to every point an option is _used_.